internal/publish/model/exposure_model_test.go

```diff
@@ -1360,11 +1420,100 @@ func TestDaysFromSymptomOnset(t *testing.T) {
 	}
 }
 
+func TestReviseKeys_FromFederation(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name     string
+		existing *Exposure
+		incoming *Exposure
+		err      error
+	}{
+		{
+			name: "export_import_mismatch",
+			existing: &Exposure{
+				ExposureKey:     []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15},
+				LocalProvenance: false,
+				ExportImportID:  proto.Int64(2),
+			},
+			incoming: &Exposure{
+				ExposureKey:    []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15},
+				ReportType:     "fake",
+				ExportImportID: proto.Int64(4),
+			},
+			err: ErrorNotSameFederationSource,
+		},
+		{
+			name: "export_import_same",
+			existing: &Exposure{
+				ExposureKey:     []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15},
+				LocalProvenance: false,
+				ReportType:      "likely",
+				ExportImportID:  proto.Int64(2),
+			},
+			incoming: &Exposure{
+				ExposureKey:    []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15},
+				ReportType:     "confirmed",
+				ExportImportID: proto.Int64(2),
+			},
+		},
+		{
+			name: "federation_mismatch",
+			existing: &Exposure{
+				ExposureKey:       []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15},
+				LocalProvenance:   false,
+				FederationQueryID: "foo",
+			},
+			incoming: &Exposure{
+				ExposureKey: []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15},
+				ReportType:  "fake",
+			},
+			err: ErrorNotSameFederationSource,
+		},
+		{
+			name: "federation_mismatch",
+			existing: &Exposure{
+				ExposureKey:       []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15},
+				LocalProvenance:   false,
+				ReportType:        "likely",
+				FederationQueryID: "foo",
+			},
+			incoming: &Exposure{
+				ExposureKey:       []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15},
+				ReportType:        "confirmed",
+				FederationQueryID: "foo",
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
Minor nit: given that you wanted to parallel the main test above, in here you could do this:
```go
tc := tc
t.Run(tc.name, func(t *testing.T) {
    t.Parallel()
```